ComponentKit/Core/Scope/CKComponentScopeHandle.mm

```diff
@@ -125,12 +142,28 @@ - (BOOL)acquireFromComponent:(CKComponent *)component
 {
   if (!_acquired && [component isMemberOfClass:_componentClass]) {
     _acquired = YES;
+    _acquiredComponent = component;
     return YES;
   } else {
     return NO;
   }
 }
 
+- (void)resolve
```

Comment:
I think it's kind of nifty. Nice way to work around the problem of controllers receiving a reference to a partially initialized component.